### PR TITLE
add AWS sigv4 support

### PR DESCRIPTION
**Pull request by penghuo**
### Description
1. add usage example. https://github.com/penghuo/os-sql/blob/flint-sigv4-pr/flint/docs/index.md#flint-datasource-readwrite
2. verified in using AWS EMR and AWS OpenSearch
    2.1. switch to Java 11
by default, EMR using Java 8. Flint Spark is compiled using Java 11.
```
sudo alternatives --config java
```
  2.2 mapping emr service role in FGAC
mapping AWS service role to FGAC role in AOS security. for example, mapping arn:aws:iam::xxx:role/service-role/AmazonEMR-InstanceProfile-yyy to all_access Roles in AOS Security

  2.3 test read AWS OpenSearch data
```
import org.apache.spark.sql.SQLContext
import org.apache.spark.sql.SQLContext._
import org.apache.spark.sql._
import org.apache.spark.sql.types._

val aos = Map("host" -> "search-mytest-00001-xxx.us-west-2.es.amazonaws.com", "port" -> "-1", "scheme" -> "https", "auth" -> "sigv4", "region" -> "us-west-2")

val sql = new SQLContext(sc)
val df = sql.read.format("flint").options(aos).schema("aInt int").load("t001")
```
3. Remove scoverage plugin
The scoverage plugin adds instrumentation code to scala source code to measure test coverage. when creating an assembly, we don't want to include this instrumentation code.
Ignore scoverage plugin for now, will add it back later.
 
### Issues Resolved
n/a
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: flint/docs/index.md [hunk 1]
```diff
@@ -171,8 +171,15 @@ In the index mapping, the `_meta` and `properties`field stores meta and schema i
...
-- `spark.flint.indexstore.location`: default is localhost
-- `spark.flint.indexstore.port`: default is 9200
+- `spark.datasource.flint.location`: default is localhost.
+- `spark.datasource.flint.port`: default is 9200.
+- `spark.datasource.flint.scheme`: default is http. valid values [http, https]
+- `spark.datasource.flint.auth`: default is false. valid values [false, sigv4]
+- `spark.datasource.flint.region`: default is us-west-2. only been used when auth=sigv4
+- `spark.datasource.flint.write.id_name`: no default value.
+- `spark.datasource.flint.write.batch_size`: default value is 1000.
+- `spark.datasource.flint.write.refresh_policy`: default value is 1000. valid values [NONE(false), IMMEDIATE(true), WAIT_UNTIL(wait_for)]
```
Comment (dai-chen) [inline flint/docs/index.md:181]: default value is false?

Comment (dai-chen) [reply flint/docs/index.md:181]: Could you update the TODO at line 75 with example too?

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

File: flint/docs/index.md (hunk 1 above)
Comment (penghuo) [reply flint/docs/index.md]: done.

**Pull request merged**